test/ocsp/helper/helper.go

```diff
@@ -0,0 +1,188 @@
+package helper
+
+import (
+	"bytes"
+	"crypto/x509"
+	"encoding/asn1"
+	"encoding/base64"
+	"encoding/pem"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"time"
+
+	"golang.org/x/crypto/ocsp"
+)
+
+var method = flag.String("method", "GET", "Method to use for fetching OCSP")
+var urlOverride = flag.String("url", "", "URL of OCSP responder to override")
+var tooSoon = flag.Int("too-soon", 76, "If NextUpdate is fewer than this many hours in future, warn.")
+var ignoreExpiredCerts = flag.Bool("ignore-expired-certs", false, "If a cert is expired, don't bother requesting OCSP.")
+var expectStatus = flag.Int("expect-status", 0, "Expect response to have this numeric status (0=good, 1=revoked)")
+
+func getIssuer(cert *x509.Certificate) (*x509.Certificate, error) {
+	if len(cert.IssuingCertificateURL) == 0 {
+		return nil, fmt.Errorf("No AIA information available, can't get issuer")
+	}
+	issuerURL := cert.IssuingCertificateURL[0]
+	resp, err := http.Get(issuerURL)
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+	body, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return nil, err
+	}
+	var issuer *x509.Certificate
+	if resp.Header.Get("Content-Type") == "application/x-pkcs7-mime" {
+		issuer, err = parseCMS(body)
+	} else {
+		issuer, err = parse(body)
+	}
+	if err != nil {
+		return nil, fmt.Errorf("from %s: %s", issuerURL, err)
+	}
+	return issuer, nil
+}
+
+func parse(body []byte) (*x509.Certificate, error) {
+	block, _ := pem.Decode(body)
+	var der []byte
+	if block == nil {
+		der = body
+	} else {
+		der = block.Bytes
+	}
+	cert, err := x509.ParseCertificate(der)
+	if err != nil {
+		return nil, err
+	}
+	return cert, nil
+}
+
+// parseCMS parses certificates from CMS messages of type SignedData.
+func parseCMS(body []byte) (*x509.Certificate, error) {
+	type signedData struct {
+		Version          int
+		Digests          asn1.RawValue
+		EncapContentInfo asn1.RawValue
+		Certificates     asn1.RawValue
+	}
+	type cms struct {
+		ContentType asn1.ObjectIdentifier
+		SignedData  signedData `asn1:"explicit,tag:0"`
+	}
+	var msg cms
+	_, err := asn1.Unmarshal(body, &msg)
+	cert, err := x509.ParseCertificate(msg.SignedData.Certificates.Bytes)
+	if err != nil {
+		return nil, fmt.Errorf("parsing CMS: %s", err)
+	}
+	return cert, nil
+}
+
+func Req(fileName string) (*ocsp.Response, error) {
```

Comment:
nit: This function is pretty long - are you open to splitting it up into some subfunctions?